superset/db_engine_specs/base.py

```diff
@@ -456,7 +456,7 @@ def apply_limit_to_sql(cls, sql: str, limit: int, database: "Database") -> str:
             )
             return database.compile_sqla_query(qry)
 
-        if LimitMethod.FORCE_LIMIT:
+        if cls.limit_method == LimitMethod.FORCE_LIMIT:
```

Comment:
This fixes a bug that is almost 3 years old! :-P

https://github.com/apache/superset/pull/4947/files#diff-c1b45a05045ffe619c4c4fcd6b3d521a1875f21865a60c32a845f30ef86b95c3R106

The bug was never triggered because we didn't have any DB engine specs using `LimitMethod.FETCH_ALL`.

Reply:
This seems like a big change, but the default `limit_method` is `LimitMethod.FORCE_LIMIT` and the only overrides are `LimitMethod.WRAP_SQL` which returns before this block is executed, so this _should_ be ok.

Should we add some tests around this?

Reply:
@robdiciuccio I'll merge so we can hotfix this, and I'll add the tests in a separate PR.

Reply:
@robdiciuccio for reference: https://github.com/apache/superset/pull/13364

Added a unit test that would've caught the bug.